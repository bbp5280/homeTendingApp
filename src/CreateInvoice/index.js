import React, { Component } from 'react';

export default class CreateInvoice extends Component{
  constructor(){
    super();
    this.state = {

    };
  }

  render () {
    return (
      <form>

      </form>
    );
  }
}
