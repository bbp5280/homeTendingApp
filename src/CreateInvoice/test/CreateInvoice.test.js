import { shallow, configure } from 'enzyme';
import  CreateInvoice from '../index';
import React from 'react';
import  Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

describe('CardCatalog snapshot', () => {

  it('should always match the snapshot', () => {
    const mkFun = jest.fn();
    const wrapper = shallow(<CreateInvoice
      home={{houseid: 1,
        message:'Message'}}
      submitInvoice={mkFun}
      close={mkFun}/>);

    expect(wrapper).toMatchSnapshot();

  });
});
