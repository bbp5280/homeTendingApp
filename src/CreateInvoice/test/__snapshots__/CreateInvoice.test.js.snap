exports[`CardCatalog snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateInvoice
    close={[Function]}
    home={
        Object {
            "houseid": 1,
            "message": "Message",
          }
    }
    submitInvoice={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="inner-modal">
        <p>
                Owner: 
                undefined undefined
        </p>
        <p>
                HouseName: 
        </p>
        <p>
                Address: 
                undefined undefined, undefined, undefined
        </p>
        <input
                onChange={[Function]}
                placeholder="Description of work"
                type="text" />
        <input
                onChange={[Function]}
                placeholder="Line Item"
                type="text"
                value="" />
        <input
                min="0.00"
                onChange={[Function]}
                placeholder="Amount"
                step="0.01"
                type="number"
                value="" />
        <button
                onClick={[Function]}>
                Add
        </button>
        <p>
                Total: 
                0
        </p>
        <button
                onClick={[Function]}>
                Cancel
        </button>
        <button
                onClick={[Function]}>
                Submit
        </button>
</form>,
      "className": "outer-modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Owner: 
            undefined undefined
</p>,
          <p>
            HouseName: 
</p>,
          <p>
            Address: 
            undefined undefined, undefined, undefined
</p>,
          <input
            onChange={[Function]}
            placeholder="Description of work"
            type="text" />,
          <input
            onChange={[Function]}
            placeholder="Line Item"
            type="text"
            value="" />,
          <input
            min="0.00"
            onChange={[Function]}
            placeholder="Amount"
            step="0.01"
            type="number"
            value="" />,
          <button
            onClick={[Function]}>
            Add
</button>,
          Array [],
          <p>
            Total: 
            0
</p>,
          <button
            onClick={[Function]}>
            Cancel
</button>,
          <button
            onClick={[Function]}>
            Submit
</button>,
        ],
        "className": "inner-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Owner: ",
              "undefined undefined",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Owner: ",
            "undefined undefined",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "HouseName: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "HouseName: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Address: ",
              "undefined undefined, undefined, undefined",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Address: ",
            "undefined undefined, undefined, undefined",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Description of work",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Line Item",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "min": "0.00",
            "onChange": [Function],
            "placeholder": "Amount",
            "step": "0.01",
            "type": "number",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Total: ",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="inner-modal">
          <p>
                    Owner: 
                    undefined undefined
          </p>
          <p>
                    HouseName: 
          </p>
          <p>
                    Address: 
                    undefined undefined, undefined, undefined
          </p>
          <input
                    onChange={[Function]}
                    placeholder="Description of work"
                    type="text" />
          <input
                    onChange={[Function]}
                    placeholder="Line Item"
                    type="text"
                    value="" />
          <input
                    min="0.00"
                    onChange={[Function]}
                    placeholder="Amount"
                    step="0.01"
                    type="number"
                    value="" />
          <button
                    onClick={[Function]}>
                    Add
          </button>
          <p>
                    Total: 
                    0
          </p>
          <button
                    onClick={[Function]}>
                    Cancel
          </button>
          <button
                    onClick={[Function]}>
                    Submit
          </button>
</form>,
        "className": "outer-modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Owner: 
              undefined undefined
</p>,
            <p>
              HouseName: 
</p>,
            <p>
              Address: 
              undefined undefined, undefined, undefined
</p>,
            <input
              onChange={[Function]}
              placeholder="Description of work"
              type="text" />,
            <input
              onChange={[Function]}
              placeholder="Line Item"
              type="text"
              value="" />,
            <input
              min="0.00"
              onChange={[Function]}
              placeholder="Amount"
              step="0.01"
              type="number"
              value="" />,
            <button
              onClick={[Function]}>
              Add
</button>,
            Array [],
            <p>
              Total: 
              0
</p>,
            <button
              onClick={[Function]}>
              Cancel
</button>,
            <button
              onClick={[Function]}>
              Submit
</button>,
          ],
          "className": "inner-modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Owner: ",
                "undefined undefined",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Owner: ",
              "undefined undefined",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "HouseName: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "HouseName: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Address: ",
                "undefined undefined, undefined, undefined",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Address: ",
              "undefined undefined, undefined, undefined",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Description of work",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Line Item",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "min": "0.00",
              "onChange": [Function],
              "placeholder": "Amount",
              "step": "0.01",
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Total: ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Total: ",
              0,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cancel",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
